nix/tests.nix

```diff
@@ -158,12 +161,85 @@ let
       ''
         env="$(cat ${postgrest.env})"
         export PATH="$env/bin:$PATH"
+        find . -type f -iname "*.tix" -delete
 
         ${withTmpDb postgresql} \
             ${cabal-install}/bin/cabal v2-run ${devCabalOptions} --verbose=0 -- \
             postgrest --dump-schema \
             | ${yq}/bin/yq -y .
       '';
+
+  coverage =
+    name: postgresql:
+    checkedShellScript
+      {
+        inherit name;
+        docs = "Run spec and io tests while collecting hpc coverage data.";
+        inRootDir = true;
+      }
+      ''
+        env="$(cat ${postgrest.env})"
+        export PATH="$env/bin:$PATH"
+
+        # clean up previous coverage data
+        rm -rf coverage && mkdir coverage
+        find . -type f -iname "*.tix" -delete
+
+        # build once before running all the tests
+        ${cabal-install}/bin/cabal v2-build ${devCabalOptions} --enable-tests all
```

Comment:
For me it's building the library and tests twice - I think it's because the later invocations don't have --enable-tests all. Should we define `coverageCabalOptions` that always has that options?

Reply:
Hm, no I don't think the `--enable-tests` is the reason for that. I had that problem at the beginning as well - it only disappeared once I changed the `postgrest.cabal` file to contain the options in the right place. The underlying reason, I think, is that some parts are not built with coverage and some are - and that causes some kind of strange rebuild cycle.

Can you try to do a `postgrest-clean` before and then rebuild?

Reply:
Multiple builds unfortunately...

Reply:
Just shooting in the dark here, but maybe you have some kind of global cabal setting with `--enable-coverage`? The cabal coverage settings always resulted in those double builds for me. Can you try adding `--disable-coverage` to all 3 cabal commands (build, exec, test)? It worked the same for me, because the hpc flags are passed manually.